controllers/network/dnsmasq_controller.go

```diff
@@ -475,9 +475,12 @@ func (r *DNSMasqReconciler) generateServiceConfigMaps(
 
 	var cfg string
 	for _, option := range instance.Spec.Options {
-		cfg += option.Key + "="
-		values := strings.Join(option.Values, ",")
-		cfg += values + "\n"
+		cfg += option.Key
+		if len(option.Values) > 0 {
+			values := strings.Join(option.Values, ",")
+			cfg += "=" + values
+		}
+		cfg += "\n"
```

Comment:
You could add a test case in https://github.com/openstack-k8s-operators/infra-operator/blob/main/tests/functional/dnsmasq_controller_test.go covering both case: i) option with values, ii) option without values

Reply:
Ah, was wondering if there were any tests... lemme see if I'm able to produce one. My golang is... well. what it is. Level-0